doc/classes/TileMap.xml

```diff
@@ -274,6 +274,12 @@
 		<member name="cell_y_sort" type="bool" setter="set_y_sort_mode" getter="is_y_sort_mode_enabled">
 			If [code]true[/code], the TileMap's children will be drawn in order of their Y coordinate. Default value: [code]false[/code].
 		</member>
+		<member name="compatibility_mode" type="bool" setter="set_compatibility_mode" getter="is_on_compatibility_mode">
+			If [code]true[/code], the compatibility with the tilemaps made in godot 3.0-3.1 is maintained (textures move when the tile origin changes and rotate if the texture size is not homogeneous). This mode presents problems when doing flip_h, flip_v and transpose in non homogeneous isometric tiles (ex 2: 1), in which the texture could not coincide with the collision, then it is not recommended for isometric tiles or not squared tiles. If [code]false[/code], the textures do not move when doing flip_h, flip_v and transpose if offset is not used, nor when changing the tile origin. Compatibility mode doesn´t work with centered textures option, because displacing textures with tile_origin option or in irregular tiles is opposite to center those textures. Default value: [code]true[/code].
```

Comment:
Will need to be rebased as we added default values in `<member>` tags now, so there is a conflict. You also don't need to specify the "Default value: [code]true[/code]" manually anymore.

Slight edit suggestion for language/formatting:
```
If [code]true[/code], the compatibility with the tilemaps made in Godot 3.1 or earlier is maintained (textures move when the tile origin changes and rotate if the texture size is not homogeneous). This mode presents problems when doing [code]flip_h[/code], [code]flip_v[/code] and [code]transpose[/code] tile operations on non-homogeneous isometric tiles (e.g. 2:1), in which the texture could not coincide with the collision, thus it is not recommended for isometric or non-square tiles.
If [code]false[/code], the textures do not move when doing [code]flip_h[/code], [code]flip_v[/code] operations if no offset is used, nor when changing the tile origin.
The compatibility mode doesn't work with the [member centered_textures] option, because displacing textures with the [code]tile_origin[/code] option or in irregular tiles is not relevant when centering those textures.
```